examples/cfg_example_oasst1.py

```diff
@@ -49,7 +50,7 @@ class Config(ConfigProblemBase):
         token_mask_probability=0.0
     )
     architecture: ConfigNLPCausalLMArchitecture = ConfigNLPCausalLMArchitecture(
-        backbone_dtype="float16",
+        backbone_dtype="float32" if torch.backends.mps.is_available() else "float16",
```

Comment:
mps doesn't support float16 currently, but support is planned and Apple announced macOS Sonoma released this fall will support AMP

https://github.com/pytorch/pytorch/issues/88415
https://github.com/pytorch/pytorch/issues/104191

Reply:
In this case, I would hide the setting on macOS using the `_visibility` flag and additionally replace any settings != float32 with a warning for CLI users